src/renderer/input.rs

```diff
@@ -339,6 +339,10 @@ pub fn imgui_wnd_proc_impl<T: RenderEngine>(
         WM_SIZE => {
             pipeline.resize(loword(lparam as u32) as u32, hiword(lparam as u32) as u32);
         },
+        WM_ACTIVATEAPP => {
+            tracing::error!("WM_ACTIVATEAPP trigered, calling reset from engine");
+            pipeline.engine.reset();
```

Comment:
Similar to above, I wouldn't report this event at all as it is explicitly necessary, and also I'd change this to be a method of `pipeline` instead, so we can avoid making the `engine` field `pub`:
```suggestion
            pipeline.reset();
```